java/vector/src/main/java/org/apache/arrow/vector/complex/DictionaryVector.java

```diff
@@ -78,13 +82,13 @@ public static DictionaryVector encode(ValueVector vector) {
 
     // copy the dictionary values into the dictionary vector
     TransferPair dictionaryTransfer = vector.getTransferPair(vector.getAllocator());
-    ValueVector dictionaryVector = dictionaryTransfer.getTo();
+    FieldVector dictionaryVector = (FieldVector) dictionaryTransfer.getTo();
     dictionaryVector.allocateNewSafe();
     for (Map.Entry<Integer, Integer> entry: transfers.entrySet()) {
       dictionaryTransfer.copyValueSafe(entry.getKey(), entry.getValue());
     }
     dictionaryVector.getMutator().setValueCount(transfers.size());
-    Dictionary dictionary = new Dictionary(dictionaryVector, false);
+    Dictionary dictionary = new Dictionary(dictionaryVector);
```

Comment:
Sorry that I've added confusion on this point. At least in the C++ implementation I don't intend to assign dictionary ids until entering the IPC code paths